src/storage/cache/rpc/error.rs

```diff
@@ -0,0 +1,42 @@
+use std::fmt::{self, Error};
+
+/// Error types for the RPC server and client
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum RpcError<T> {
+    /// The request is invalid.
+    InvalidRequest(T),
+    /// The response is invalid.
+    InvalidResponse(T),
+    /// The server/client meet an internal error.
+    InternalError(T),
+}
```

Comment:
It would be better remote T from generic,  since all the places use String.
add Timeout for convienent

Reply:
OK.